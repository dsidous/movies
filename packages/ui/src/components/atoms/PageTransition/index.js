import React, { useState, useEffect } from 'react';
import { CSSTransition } from 'react-transition-group';

import { propTypes } from './propTypes';

const PageTransition = props => {
  const { children } = props;
  const [loaded, setLoaded] = useState(false);

  useEffect(() => {
    setLoaded(true);
  }, []);

  return (
    <CSSTransition in={loaded} timeout={1000} classNames="fade">
      <div>{children}</div>
    </CSSTransition>
  );
};

PageTransition.propTypes = propTypes;

export default PageTransition;
