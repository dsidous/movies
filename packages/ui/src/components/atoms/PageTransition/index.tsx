/** @jsxFrag React.Fragment */
import React, { useState, useEffect } from 'react';
import { CSSTransition } from 'react-transition-group';

import useStyles from './styles';

interface Props {
  children: React.ReactNode;
}

const PageTransition:React.FC<Props> = ({ children }) => {
  const classes = useStyles();

  const [loaded, setLoaded] = useState<boolean>(false);

  useEffect(() => {
    setLoaded(true);
  }, []);

  return (
    <CSSTransition
      in={loaded}
      timeout={1000}
      classNames={{
        appear: classes.appear,
        appearActive: classes.appearActive,
        enter: classes.enter,
        enterActive: classes.enterActive,
      }}
    >
      <>{children}</>
    </CSSTransition>
  );
};

export default PageTransition;
