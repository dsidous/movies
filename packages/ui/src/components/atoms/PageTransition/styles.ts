import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  {
    appear: {
      opacity: 0.01,
    },
    enter: {
      opacity: 0.01,
    },

    appearActive: {
      opacity: 1,
      transition: 'opacity 0.2s ease-in',
    },

    enterActive: {
      opacity: 1,
      transition: 'opacity 0.2s ease-in',
    },
  },
  { name: 'fade' },
);
