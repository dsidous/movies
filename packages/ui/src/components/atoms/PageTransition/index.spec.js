import React from 'react';
import { shallow } from 'enzyme';

import PageTransition from '.';

describe('Atoms/PageTransition', () => {
  it('should render as expected', () => {
    const wrapper = shallow(
      <PageTransition>
        <div />
      </PageTransition>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
