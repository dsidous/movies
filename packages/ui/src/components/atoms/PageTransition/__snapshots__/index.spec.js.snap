// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/PageTransition should render as expected 1`] = `ShallowWrapper {}`;
